tests/test_route_map.rb

```diff
@@ -0,0 +1,1462 @@
+  false
+end
+
+# TestRouteMap - Minitest for RouteMap
+# node utility class
+class TestRouteMap < CiscoTestCase
+  @skip_unless_supported = 'route_map'
+
+  def setup
+    super
+  end
+
+  def teardown
+    config_no_warn('no route-map MyRouteMap permit 100')
+    super
+  end
+
+  def create_route_map(name='MyRouteMap', seq='100', action='permit')
+    RouteMap.new(name, seq, action)
+  end
+
+  def test_description
+    rm = create_route_map
+    assert_equal(rm.default_description, rm.description)
+    rm.description = 'TestRouteMap'
+    assert_equal('TestRouteMap', rm.description)
+    rm.description = rm.default_description
+    assert_equal(rm.default_description, rm.description)
+  end
+
+  def test_collection_size
+    rm1 = create_route_map('my1', '100', 'permit')
+    rm2 = create_route_map('my1', '101', 'permit')
+    rm3 = create_route_map('my1', '102', 'deny')
+    rm4 = create_route_map('my2', '102', 'permit')
+    rm5 = create_route_map('my2', '104', 'deny')
+    rm6 = create_route_map('my3', '105', 'deny')
+    assert_equal(1, RouteMap.maps['my1']['100'].size)
+    assert_equal(1, RouteMap.maps['my1']['101'].size)
+    assert_equal(1, RouteMap.maps['my1']['102'].size)
+    assert_equal(3, RouteMap.maps['my1'].size)
+    assert_equal(1, RouteMap.maps['my2']['102'].size)
+    assert_equal(1, RouteMap.maps['my2']['104'].size)
+    assert_equal(2, RouteMap.maps['my2'].size)
+    assert_equal(1, RouteMap.maps['my3']['105'].size)
+    assert_equal(1, RouteMap.maps['my3'].size)
+    rm1.destroy
+    rm2.destroy
+    rm3.destroy
+    rm4.destroy
+    rm5.destroy
+    rm6.destroy
+  end
+
+  def test_match_as_number
+    rm = create_route_map
+    assert_equal(rm.default_match_as_number, rm.match_as_number)
+    array = ['3', '22-34', '35', '100', '101-220']
+    rm.match_as_number = array
+    assert_equal(array, rm.match_as_number)
+    rm.match_as_number = rm.default_match_as_number
+    assert_equal(rm.default_match_as_number, rm.match_as_number)
+  end
+
+  def test_match_as_number_as_path_list
+    rm = create_route_map
+    assert_equal(rm.default_match_as_number_as_path_list, rm.match_as_number_as_path_list)
+    array = %w(cbc xyz pwd)
+    rm.match_as_number_as_path_list = array
+    assert_equal(array, rm.match_as_number_as_path_list)
+    rm.match_as_number_as_path_list = rm.default_match_as_number_as_path_list
+    assert_equal(rm.default_match_as_number_as_path_list, rm.match_as_number_as_path_list)
+  end
+
+  def test_community
+    rm = create_route_map
+    assert_equal(rm.default_match_community, rm.match_community)
+    assert_equal(rm.default_match_community_exact_match, rm.match_community_exact_match)
+    array = %w(public private)
+    rm.match_community_set(array, false)
+    assert_equal(array, rm.match_community)
+    assert_equal(false, rm.match_community_exact_match)
+    rm.match_community_set(array, true)
+    assert_equal(array, rm.match_community)
+    assert_equal(true, rm.match_community_exact_match)
```

Comment:
Make sure to use more specific assertion instead of assert_equal if possible.

https://github.com/cisco/cisco-network-node-utils/blob/develop/docs/README-develop-best-practices.md#mt2